packages/polling-controller/src/PollingController.ts

```diff
@@ -6,6 +6,11 @@ import { v4 as random } from 'uuid';
 // See TS2545
 type Constructor = new (...args: any[]) => object;
 
+// get unique key for outer poll token
+// this is used to group networkClientId polls with the same options
+export const getKey = (networkClientId: NetworkClientId, options: any) =>
+  `${networkClientId}:${JSON.stringify(Object.entries(options).sort())}`;
```

Comment:
Instead of sorting keys and running them through `JSON.stringify`, what do you think about using [`fast-json-stable-stringify`](https://www.npmjs.com/package/fast-json-stable-stringify)? It's already in the dependency tree and is used for e.g. `@metamask/snaps-utils`.

Reply:
does `fast-json-stable-stringify` sort by default? i still need to preserve the order, since objects aren't guarenteed the same order in js

Reply:
looks like [it does](https://www.npmjs.com/package/fast-json-stable-stringify), or you can pass a preferred sorting function